### PR TITLE
Fix: Sparce Tensor wrong exeption Message when passing argument with wrong Type

### LINKED ISSUE #37640: Sparce Tensor wrong exeption Message when passing argument with wrong Type
System information

    Have I written custom code (as opposed to using a stock
    example script provided in TensorFlow): No
    OS Platform and Distribution (e.g.,
    Linux Ubuntu 16.04): Linux Mint 19.3 (ubuntu)
    TensorFlow installed from (source or
    binary): binatry
    TensorFlow version (use command below): both tf-nightly and tf-2.1
    Python version: 3.8 and 3.6

**Describe the current behavior**
when passing an int32 tensor to Sparce Tensor
last Error displayed is:
```
ValueError: Unable to create eager SparseTensor. Check that your shape is correctly defined. Eager SparseTensors don't support unknown dimesions.
got shape:
    [4 4 4 4]
```
when looking back in stack trace, the right error is raced:

`ValueError: Tensor conversion requested dtype int64 for Tensor with dtype int32: <tf.Tensor: shape=(4,), dtype=int32, numpy=array([4, 4, 4, 4], dtype=int32)>`

**Describe the expected behavior**
one of the following:
1) conversion should not fail
2) last Error should be:
expected int64 tensor for shape argument got int32 tensor

**Standalone code to reproduce the issue** 
```
import tensorflow as tf

indices = tf.cast([[1,1,1,1],[1,3,1,1]],dtype=tf.int64)
shape = tf.cast([4,4,4,4],dtype=tf.int64)

heat_map = tf.SparseTensor(indices = indices, values = tf.ones(tf.shape(indices)[0]), dense_shape = shape)

indices = tf.cast([[1,1,1,1],[1,3,1,1]],dtype=tf.int64)
shape = tf.cast([4,4,4,4],dtype=tf.int32)

heat_map = tf.SparseTensor(indices = indices, values = tf.ones(tf.shape(indices)[0]), dense_shape = shape)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

```
Traceback (most recent call last):
  File "/home/bhb/Cloud/Code/Git/3D_Person_Pose_Estimation_from_2D_Singelview_Image_Data/src/venv/lib/python3.6/site-packages/tensorflow/python/framework/sparse_tensor.py", line 142, in __init__
    dense_shape, name="dense_shape", dtype=dtypes.int64)
  File "/home/bhb/Cloud/Code/Git/3D_Person_Pose_Estimation_from_2D_Singelview_Image_Data/src/venv/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1317, in convert_to_tensor
    (dtype.name, value.dtype.name, value))
ValueError: Tensor conversion requested dtype int64 for Tensor with dtype int32: <tf.Tensor: shape=(4,), dtype=int32, numpy=array([4, 4, 4, 4], dtype=int32)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bhb/.vscode/extensions/ms-python.python-2020.2.64397/pythonFiles/ptvsd_launcher.py", line 48, in <module>
    main(ptvsdArgs)
  File "/home/bhb/.vscode/extensions/ms-python.python-2020.2.64397/pythonFiles/lib/python/old_ptvsd/ptvsd/__main__.py", line 432, in main
    run()
  File "/home/bhb/.vscode/extensions/ms-python.python-2020.2.64397/pythonFiles/lib/python/old_ptvsd/ptvsd/__main__.py", line 316, in run_file
    runpy.run_path(target, run_name='__main__')
  File "/usr/lib/python3.6/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/usr/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/bhb/Cloud/Code/Git/3D_Person_Pose_Estimation_from_2D_Singelview_Image_Data/src/test_sparce_tensor.py", line 11, in <module>
    heat_map = tf.SparseTensor(indices = indices, values = tf.ones(tf.shape(indices)[0]), dense_shape = shape)
  File "/home/bhb/Cloud/Code/Git/3D_Person_Pose_Estimation_from_2D_Singelview_Image_Data/src/venv/lib/python3.6/site-packages/tensorflow/python/framework/sparse_tensor.py", line 148, in __init__
    "got shape:\n    {}".format(dense_shape))
ValueError: Unable to create eager SparseTensor. Check that your shape is correctly defined. Eager SparseTensors don't support unknown dimesions.
got shape:
    [4 4 4 4]
Beendet
```

### PR DESCRIPTION
**Pull request by rushabh-v**
resolves #37640

### CODE REVIEW

**Review by alextp — DISMISSED** [review]

**Comment by gbaned** [comment]
@rushabh-v Can you please check build failures. Thanks!

**Comment by rushabh-v** [comment]
Can you take a look again, please? @gbaned

File: tensorflow/python/framework/sparse_tensor.py [hunk 1]
```diff
@@ -141,8 +141,11 @@ def __init__(self, indices, values, dense_shape):
           dense_shape = ops.convert_to_tensor(
               dense_shape, name="dense_shape", dtype=dtypes.int64)
           dense_shape_default = tensor_shape.TensorShape(dense_shape)
-        except ValueError:
-          raise ValueError("Unable to create eager SparseTensor. Check that "
+        except ValueError as e:
+          if "Tensor conversion requested dtype int64" in str(e):
```
Comment (alextp) [inline tensorflow/python/framework/sparse_tensor.py]: I really don't like this type of error message parsing. This is super fragile.

**Review by alextp — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@rushabh-v Can you please check reviewer comments and keep us posted. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
@rushabh-v Any update on this PR, please. Thanks!

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**